src/libostree/ostree-repo.h

```diff
@@ -659,6 +683,27 @@ gboolean ostree_repo_export_tree_to_archive (OstreeRepo                *self,
                                              GCancellable             *cancellable,
                                              GError                  **error);
 
+/**
+ * OstreeRepoExportArchiveOptions2:
+ *
+ * An extensible options structure controlling archive creation.  Ensure that
+ * you have entirely zeroed the structure, then set just the desired
+ * options.  This is used by ostree_repo_export_tree_to_archive2().
+ */
+typedef struct {
+  gboolean disable_xattrs;
+  guint64 timestamp_secs;
+  char *path_prefix;
+} OstreeRepoExportArchiveOptions2;
+
+_OSTREE_PUBLIC
+gboolean ostree_repo_export_tree_to_archive2 (OstreeRepo                *self,
+                                             OstreeRepoExportArchiveOptions2  *opts,
+                                             OstreeRepoFile            *root,
+                                             void                      *archive,  /* Really struct archive * */
```

Comment:
Same issue here.  In general we're probably going to need something like `int fd` + "some form of archive options" to be introspectable.